xbmc/guilib/GUIFontTTF.cpp

```diff
@@ -745,8 +745,14 @@ void CGUIFontTTF::DrawTextInternal(CGraphicContext& context,
 
 float CGUIFontTTF::GetTextWidthInternal(const vecText& text)
 {
+  const std::u32string key(text.begin(), text.end());
+  if (m_textWidthCache.contains(key))
+    return m_textWidthCache[key];
+
   const std::vector<Glyph> glyphs = GetHarfBuzzShapedGlyphs(text);
-  return GetTextWidthInternal(text, glyphs);
+  const float width = GetTextWidthInternal(text, glyphs);
+  m_textWidthCache[key] = width;
```

Comment:
this is causing a memory leak, you are storing any kind of text in memory
if there are long texts in the long run it can be very expensive

to take in account that if you often use the keyboard to enter text or numbers in the GUI this method is called many times and you are storing an infinite number of data in to memory

a quick test kodi raised memory to 300mb~ occupied and almost 9000 items in the map never released, just to navigate in to kodi settings and used keyboard to input text

Reply:
I agree, a cache with unlimited capacity seems not a wise decision for this use case. And have we statistics for the cache hit ratio for some standard Kodi use cases? Is it actually increasing UX experience of the user? At least I was not aware of any visual improvements while testing this PR.

Reply:
iirc it was mentioned on slack time ago, i think this PR is trying to improve the problem where the GUI text scrolling like "plot" is stuttering on some low end devices(?) or something like so...but it was not mentioned in the PR description

Reply:
I agree that it's not a proper fix. At first I thought this was only happening on slower devices like the Shield TV but I installed arctic horizon 2 on my mac and got the same lag there as well. I don't think in this particular case some text is scrolling but switching from the movie to the shows tab in that skin is supposed to fade in. The whole fade in animation is not visible due to the lag caused by the font width calculation though. With this bandaid fix applied it fades in smoothly, but at a severe memory cost. I was wondering if font width could not be calculated differently by caching it per glyph which should lower memory footprint significantly.

Below is the flamegraph showing the issue
![flamegraph](https://github.com/user-attachments/assets/462a129d-378e-43bb-84cd-f71f159e6700)

Reply:
yeah the problem needs to be approached differently
but atm i dont have the complete situation at my hand

by following your call path seem that in your use cases following condition it does not seem to determine that the text is always the same
https://github.com/xbmc/xbmc/blob/835b1a9b480eafb64617e78a04b432c1dddc85bf/xbmc/guilib/GUITextLayout.cpp#L224-L225
therefore calls all methods continuously, perhaps the situation can be improved within CGUITextLayout (?)

Reply:
It should be easily reproducible by installing the arctic horizon 2 skin and just switching between the movie and shows tab on the home screen